Math/NumberTheory/Moduli/Chinese.hs

```diff
@@ -8,18 +8,120 @@
 --
 -- Chinese remainder theorem
 --
-
 {-# LANGUAGE BangPatterns #-}
 
 module Math.NumberTheory.Moduli.Chinese
-  ( chineseRemainder
+  ( chineseRemainders
+  , chineseRemainders2
+  , chineseRemainder
   , chineseRemainder2
-  ) where
+  )
+where
 
+import Data.Ratio (numerator, denominator)
 import Control.Monad (foldM)
+
 import GHC.Integer.GMP.Internals
 
 import Math.NumberTheory.GCD (extendedGCD)
+import Math.NumberTheory.Moduli.Class
+import Math.NumberTheory.Primes.Factorisation (factorise)
+
+-- [Unsure where to put these.  Does arithmoi keep QuickCheck tests in separate documents?
```

Comment:
Yes, put tests into https://github.com/CarlEdman/arithmoi/blob/patch-1/test-suite/Math/NumberTheory/Moduli/ChineseTests.hs

You can continue editing files in https://github.com/CarlEdman/arithmoi/blob/patch-1 branch, changes will appear here automatically.